File: 단계별로 풀어보기/반복문/더하기 사이클/더하기 사이클.cpp
#include <iostream>

using namespace std;

int makeNewNum(int OldNum) {
	int tenD, oneD, temp, NewNum;

	//Divide by digits
	tenD = OldNum / 10;
	oneD = OldNum % 10;

	//Make new number;
	temp = tenD;
	tenD = oneD;
	oneD = (temp + oneD)%10;
	NewNum = tenD * 10 + oneD;

	return NewNum;
}

int getCycle() {
	int num, oldNum, newNum, cycle;
	cin >> num;
	oldNum = num;
	newNum = -1;
	cycle = 1;

	newNum = makeNewNum(oldNum);
	while (num != newNum) {
		oldNum = newNum;
		newNum = makeNewNum(oldNum);
		cycle++;
	}

	return cycle;
}


int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int cycle;

	cycle = getCycle();
	cout << cycle;
}

File: 단계별로 풀어보기/반복문/더하기 사이클/더하기 사이클.py
import sys

def makeNewNum(num):
    tenDigit=num//10
    oneDigit=num%10

    temp=tenDigit
    tenDigit=oneDigit
    oneDigit=(temp+oneDigit)%10

    return tenDigit*10+oneDigit
    
def getCycle(num):
    cycle = 1
    oldNum=num
    newNum=-1
    newNum=makeNewNum(oldNum)
    
    while(num != newNum):
        oldNum=newNum
        newNum=makeNewNum(oldNum)
        cycle=cycle+1

    return cycle

def init():
    num=int(input())
    print(getCycle(num))

init()
